projects/packages/my-jetpack/src/class-initializer.php

```diff
@@ -778,9 +788,32 @@ public static function add_red_bubble_alerts( array $red_bubble_slugs ) {
 	 * @param array $red_bubble_slugs - slugs that describe the reasons the red bubble is showing.
 	 * @return array
 	 */
-	public static function alert_if_missing_site_connection( array $red_bubble_slugs ) {
-		if ( ! ( new Connection_Manager() )->is_connected() ) {
-			$red_bubble_slugs[ self::MISSING_SITE_CONNECTION_NOTIFICATION_KEY ] = null;
+	public static function alert_if_missing_connection( array $red_bubble_slugs ) {
+		$broken_modules = self::check_for_broken_modules();
+
+		if ( ! empty( $broken_modules['needs_user_connection'] ) ) {
+			$red_bubble_slugs[ self::MISSING_CONNECTION_NOTIFICATION_KEY ] = array(
+				'type'     => 'user',
+				'is_error' => true,
+			);
+		} elseif ( ! empty( $broken_modules['needs_site_connection'] ) ) {
+			$red_bubble_slugs[ self::MISSING_CONNECTION_NOTIFICATION_KEY ] = array(
+				'type'     => 'site',
+				'is_error' => true,
+			);
+		} elseif (
+			! ( new Connection_Manager() )->is_user_connected() &&
+			! ( new Connection_Manager() )->has_connected_owner()
+		) {
+			$red_bubble_slugs[ self::MISSING_CONNECTION_NOTIFICATION_KEY ] = array(
+				'type'     => 'user',
+				'is_error' => false,
+			);
+		} elseif ( ! ( new Connection_Manager() )->is_connected() ) {
+			$red_bubble_slugs[ self::MISSING_CONNECTION_NOTIFICATION_KEY ] = array(
+				'type'     => 'site',
+				'is_error' => false,
+			);
```

Comment:
This is not a big issue here, but since each case is independent, I'd recommend having early returns instead of `elseif`s to make it more readable and maintainable.

Reply:
This is a good improvement, thank you. I agree since we aren't doing anything but setting a single red bubble slug in this function currently